src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -641,12 +645,25 @@ class GoogleExtensions(
     )
   }
 
-  def getSignedUrl(samUser: SamUser, project: GoogleProject, bucket: GcsBucketName, name: GcsBlobName, samRequestContext: SamRequestContext): IO[URL] =
+  def getSignedUrl(
+      samUser: SamUser,
+      project: GoogleProject,
+      bucket: GcsBucketName,
+      name: GcsBlobName,
+      duration: Option[Long],
+      samRequestContext: SamRequestContext
+  ): IO[URL] =
     for {
       petServiceAccount <- createUserPetServiceAccount(samUser, project, samRequestContext)
       petKey <- googleKeyCache.getKey(petServiceAccount)
       serviceAccountCredentials = ServiceAccountCredentials.fromStream(new ByteArrayInputStream(petKey.getBytes()))
-      url <- googleStorageService.getSignedBlobUrl(bucket, name, serviceAccountCredentials).compile.lastOrError
+      time = duration.getOrElse(1L)
+      timeUnit = duration.map(_ => TimeUnit.MINUTES).getOrElse(TimeUnit.HOURS)
```

Comment:
This feels a little confusing no?  Why not just have
```suggestion
      time = duration.getOrElse(60L)
      timeUnit = TimeUnit.MINUTES
```

and timeUnit could just be a constant

Reply:
Good point. Inlined the timeUnit and renamed the variable to be clearer.